pkg/api/datasources.go

```diff
@@ -597,7 +597,13 @@ func getEncodedString(jsonData *simplejson.Json, key string) string {
 }
 
 func checkTeamHTTPHeaderPermissions(hs *HTTPServer, c *contextmodel.ReqContext, ds *datasources.DataSource, cmd datasources.UpdateDataSourceCommand) (bool, error) {
-	currentTeamHTTPHeaders := getEncodedString(ds.JsonData, "teamHttpHeaders")
+	if cmd.JsonData == nil {
```

Comment:
@alexanderzobnin had to check for existence of the jsondata for both cases if the ds do not have it, or if the cmd has it.

Could you review this?

Reply:
Why not to check it inside `getEncodedString`?

Reply:
Because the ds.JsonData in some cases are `nil` when we have not any jsonData. Then we want to access something that is nil in the `getEncodedString`

Reply:
i reread your comment. Il check it inside the `getEncodedString` .

Is this to compartmentalize the checks as close to the source as possible? 🤔